### ISSUE TITLE
Check for reassignment before use.

### ISSUE THREAD

**Issue by kisielk** [kisielk/errcheck#7]
It would be nice to catch code such as:

```
var (
    x, y int
    err error
)
x, err = foo()
y, err = a(x)
return x, y
```

This is not an error according to the compiler: http://play.golang.org/p/3DJj2e3-8L
The upcoming ssa package would likely be useful here.

**Comment by dmage** [comment]
Also it would be nice to catch unhandled error in this example:
```
package main

import "net/http"

func main() {
	resp1, err := http.Get("1")
	resp1 = resp1

	resp2, err := http.Get("2")
	if err != nil {
		panic(err)
	}
	resp2 = resp2
}
```

**Comment by kisielk** [comment]
I agree. Will have to learn about using the SSA package to figure out how
to do this. If you have any ideas or examples any help would be welcome.

**Comment by mhamrle** [comment]
this tool does something similar https://github.com/gordonklaus/ineffassign . I can catch example from first comment.

**Comment by tamird** [comment]
https://github.com/dominikh/go-staticcheck catches this sort of thing, FWIW.

**Comment by echlebek** [comment]
@kisielk should we close this, given that staticcheck has it covered?

**Comment by kisielk** [comment]
I think it still falls in the scope of this package, maybe it will get added at some point

**Comment by prestonvanloon** [comment]
Here's a test case for potential reassignment: https://github.com/kisielk/errcheck/commit/8320fd2f
